### ISSUE TITLE
Last data is missassigned in a time series when the filter its too long

### ISSUE THREAD

**Issue by manolopm** [grafana/grafana#32972]
**What happened**:

I have a time series with a value per day. If I took a short period (like last 7 days) all the data appears correctly. In the example, for today (2021-04-14 00:00:00) appears 453 as value for that day. If I took a longer period (like last 2 years) appears with another value assigned to 2021-04-13 13:00:00 . I have UTC+1 timezone, I think it's not related to timezone but who knows...

**What you expected to happen**:

That the value that its assigned to 2021-04-14 00:00:00 (1618315200000 in the database as ts) appears in that datetime in short ranges filter and also in the same datetime on longer ones.

**How to reproduce it (as minimally and precisely as possible)**:

This is my query :

SELECT
  $__timeGroupAlias(day,$__interval),
sum(recargas85*8.5+recargas10*10+recargas15*15+recargas20*20+recargas25*25+recargas30*30+recargas35*35+recargas40*40+recargas45*45+recargas50*50) AS "Cantidad"
FROM dm_portalrecargas_ganancias
WHERE
  $__timeFilter(day)
GROUP BY 1
ORDER BY 1

Just add data (sorry but I cannot provide mines) with at least 6 months every day and try to look it with 7  days filter and also with 2 years filter.

**Anything else we need to know?**:

I take a look into query inspector. The last values in raw are different in quantity and timestamp values. 

For two years filter:
![image](https://user-images.githubusercontent.com/60767/114678775-7b0f6000-9d03-11eb-811e-37669e9bd839.png)

For 7 days filter:
![image](https://user-images.githubusercontent.com/60767/114678847-8e223000-9d03-11eb-8de5-715841354e14.png)

Database source:
![image](https://user-images.githubusercontent.com/60767/114679670-49e35f80-9d04-11eb-9786-9e504647cc1a.png)

**Environment**:
- Grafana version: 7.4.3 Debian packaged
- Data source type & version: Postgresql-11 11.8-1.pgdg100+1 Debian packaged
- OS Grafana is installed on: Debian 10
- User OS & Browser: Windows 10 & Chrome
- Grafana plugins: 
- Others: GIF With the problem
![grafana](https://user-images.githubusercontent.com/60767/114680387-06d5bc00-9d05-11eb-9683-8650830d0ea6.gif)

**Comment by manolopm** [comment]
It doesn't seems to be something from frontend.

I have checked the query urls with chrome inspector and get the query from and to parameters, then I get the timestamp of the last value in the response and this is what I get:

![image](https://user-images.githubusercontent.com/60767/114857768-913f1e00-9de0-11eb-8ffa-a599a3a58c12.png)

As you can see the first request its with only one week and the second its with two years range and both return diferent last time with eleven hours of difference.

I grab the urls and put it into a curl request and launched it again and again with the same parameters with the same result. The query for a week gives me the last in April 15th 2021 at 00:00:00.000 (in Canary Island timezone GTM+1) , and this is the right timestamp, and the two years request gives me April 14th 2021 at 13:00:00.000 (Again in my timezone).

I'll try to debug the endpoint to check where seems to be changing the timestamp. If I found anything I'll report it here.

**Comment by manolopm** [comment]
I've been taken a look to the query inspector in both cases and I think the problem is in the query itself.

With one year so far lapse the query shows this:

![image](https://user-images.githubusercontent.com/60767/115976022-a266f800-a561-11eb-9142-0ba9122a4551.png)

And with last 7 days query show this:

![image](https://user-images.githubusercontent.com/60767/115976037-ce827900-a561-11eb-976f-d94301313fc3.png)

I've notice time is calculated dividing by a number and multiplying by the same number. This number changes in both querys (from 7200 to 300). 
I took both querys into postgres directly and I have the same problem , but I show day without any processing also and it was the same value .

I have removed the group by that I had in my query ( group by ) and seems to work as It should. 

Maybe it was a problem in my query (and this should be closed if its a problem in my query), but still don't understand why make something strange grouping by days and so on.
